object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionTransfer.java

```diff
@@ -71,7 +71,8 @@ public ObjectConstructionTransfer(final CFAnalysis analysis) {
     CFStore thenStore = result.getThenStore();
     CFStore elseStore = result.getElseStore();
 
-    while (receiver != null) {
+    while (receiver != null && !receiver.toString().startsWith("(this).")) {
```

Comment:
Not sure how much we care or how much it matters in this case, but I think these `toString()` calls can be bad for checker performance

Reply:
I agree we should avoid that.  String comparisons also tend to be brittle (such as depending on the number of parentheses).